src/server/saved_objects/service/lib/repository.js

```diff
@@ -36,6 +36,7 @@ export class SavedObjectsRepository {
       schema,
       serializer,
       migrator,
+      allowedTypes = [],
```

Comment:
Changing this would create inconsistent naming and further confuse the usage of `types` in repository which is already convoluted. `allowedTypes` says that the repository will restrict all the methods to this set of types. Imho is more explicit since the repository is known just as an adapter to ES which I see as more of the source of the types. Using `types` would be ambiguous as to the purpose of the list within the repository and what that list changes about the repository.